### PR TITLE
mysql_user: add functionality to disable binlog when working with users

### PR DESCRIPTION
**Pull request by grypyrg**
mysql_user: add functionality to disable binlog when working with users.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @grypyrg. @marktheunissen please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by marktheunissen** [comment]
I am no longer the maintainer, as per https://github.com/ansible/ansible-modules-core/pull/2653

**Comment by robynbergeron** [comment]
Thanks @grypyrg for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by tersmitten** [comment]
Nice addition. I think the problem is also present in the `mysql_db` module.

File: database/mysql/mysql_user.py [hunk 1]
```diff
@@ -204,7 +214,13 @@ def connect(module, login_user=None, login_password=None, config_file=''):
         config['passwd'] = login_password
 
     db_connection = MySQLdb.connect(**config)
-    return db_connection.cursor()
+
+    cursor = db_connection.cursor()
+
+    if module.params['sql_log_bin'] == 'no':
```
Comment (tersmitten) [inline database/mysql/mysql_user.py:220]: What about `true`/`false` and `1`/`0`?

File: database/mysql/mysql_user.py [hunk 2]
```diff
@@ -395,6 +411,7 @@ def main():
             check_implicit_admin=dict(default=False, type='bool'),
             update_password=dict(default="always", choices=["always", "on_create"]),
             config_file=dict(default="~/.my.cnf"),
+            sql_log_bin=dict(default="yes")
```
Comment (tersmitten) [inline database/mysql/mysql_user.py:414]: Wouldn't `default=True, type=bool` be better?

**Comment by Jmainguy** [comment]
@grypyrg @tersmitten So right now by default it always logs to bin, and this allows one to change to false, in order not to log stuff?

The benefit being gives the user the choice to free up some disk space?

**Comment by tersmitten** [comment]
No it's not really because of space. It's because sometimes you don't want to log queries to the binlog because they may break replication (if you run them multiple times, e.g. master-master) or you don't mind to create a difference (in users for instance) between machines

**Comment by Jmainguy** [comment]
@tersmitten Gotcha that makes sense

@grypyrg Would you mind rebasing?

**Comment by Jmainguy** [comment]
@grypyrg Think you could rebase this?

**Comment by grypyrg** [comment]
@Jmainguy : I'd like to but I don't have sufficient time at the moment to make a fix. I can try to do this in April, where I might have some spare time.

**Comment by Jmainguy** [comment]
@grypyrg No worries, I know how busy life can get. I will take a stab at rebasing this later this week (which should let the PR to get merged, and you will retain credit). I will update here once I have attempted it.

**Comment by Jmainguy** [comment]
Rebased over at #3112 I recommend closing this out, and continuing discussion over there.

**Comment by sivel** [comment]
Closing per above.

**Pull request closed without merging**